src/main/java/baritone/map/MapChunk.java

```diff
@@ -0,0 +1,98 @@
+package baritone.map;
+
+import baritone.bot.Baritone;
+import baritone.bot.utils.BlockStateInterface;
+import baritone.bot.utils.pathing.BetterBlockPos;
+import net.minecraft.block.material.MapColor;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockPos.MutableBlockPos;
+import net.minecraft.world.chunk.Chunk;
+import org.lwjgl.util.Color;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class MapChunk {
+
+    private Chunk chunk;
+
+    public MapChunk(Chunk chunk) {
+        this.chunk = chunk;
+    }
+
+    public void writeImage() {
+        Path image = getImagePath();
+        if(!Files.exists(image.getParent())) {
+            try {
+                Files.createDirectory(image.getParent());
```

Comment:
Redundant with the createDirectories call in Baritone.java, no? Probably just this one would work

Reply:
No, this is creating the "map" directory.